### PR TITLE
[RHELC-1604] Verify grub file validity check

### PR DESCRIPTION
**Pull request by danmyway**
* move the grub file with invalid changes test to a non-destructive suite, validate the error is raised in such scenario

Jira Issues:

- [RHELC-1604](https://issues.redhat.com/browse/RHELC-1604)

Checklist

- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` or `[HMS-]` is part of the PR title 
- [x] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: tests/integration/tier0/non-destructive/grub/main.fmf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    - sanity
+    - grub
+    - grub-mkfile
+
+/invalid_changes_to_grub_file:
+    summary+: |
+        Introduce invalid changes
+    description+: |
+        Invalid changes, e.g. removing the `=` should cause the 'grub2-mkfile' call to fail.
+        Verify a successful conversion.
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/grub/main.fmf:17]:
```suggestion
```

File: tests/integration/tier0/non-destructive/grub/test_invalid_changed_grub.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import pytest
+
+from conftest import TEST_VARS
+
+
+target_line = "GRUB_CMDLINE_LINUX"
+
+
+@pytest.fixture
+def grub_file_invalid(shell_fixture, backup_directory):
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/grub/test_invalid_changed_grub.py:15]: why `shell_fixture` instead of just `shell`?

**Review by kokesak — CHANGES_REQUESTED** [review]

File: tests/integration/tier0/non-destructive/grub/test_invalid_changed_grub.py (hunk 2 above)
Comment (danmyway) [reply tests/integration/tier0/non-destructive/grub/test_invalid_changed_grub.py:15]: That might be a result of auto-completion and laziness :no_mouth: 
Good catch though

**Review by kokesak — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**